PowerValidatedSolutions.psm1

```diff
@@ -50441,6 +50441,469 @@ Function Test-WMSubnetInput {
+        [Parameter (Mandatory = $true, ParameterSetName = 'msca')] [ValidateNotNullOrEmpty()] [String]$caFqdn,
+        [Parameter (Mandatory = $true, ParameterSetName = 'msca')] [ValidateNotNullOrEmpty()] [String]$username,
+        [Parameter (Mandatory = $true, ParameterSetName = 'msca')] [ValidateNotNullOrEmpty()] [String]$password,           
+        [Parameter (Mandatory = $true, ParameterSetName = 'msca')] [ValidateNotNullOrEmpty()] [String]$certificateTemplate,
+        [Parameter (Mandatory = $false, ParameterSetName = 'msca')] [Switch]$getCArootCert
+    )
+    
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDirPath)) {
+        Write-Error "Certificate directory $certDirPath does not exist: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    }
+    if (-Not (Test-Path -Path $csrFilePath)) {
+        Write-Error "Certificate signing request file ($csrFilePath) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    }
+    $fileName = Split-Path -Path $csrFilePath -LeafBase
+    $crtFileName = $fileName + ".crt"
+    $crtFilePath = Join-Path -Path $certDirPath -ChildPath $crtFileName
+    if (Test-Path -Path $crtFilePath) {
+        Write-Error "$crtFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+    }
+    if (-Not ([bool] (Get-Command -ErrorAction Ignore -Type Application openssl))) {
+        Write-Error "OpenSSL not found.  Please verify OpenSSLis installed: PRE_VALIDATION_FAILED" -ErrorAction Stop
+    } else {
+        $opensslVersion = openssl version | ForEach-Object { ($_ -split ' ')[1] }
+        if ($opensslVersion -lt "3.0.0") {
+            Write-Error "OpenSSL version is less than 3.0: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+    }
+
+    if($certificateAuthority -eq "openssl") {
+        # validate inputs
+        if (-Not (Test-Path -Path $caKeyPath)) {
+            Write-Error "Openssl CA private key file ($caKeyPath) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+        if (-Not (Test-Path -Path $caCertPath)) {
+            Write-Error "Openssl CA certificate file ($caCertPath) not found: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+        if ($expireDays -lt 2) {
+            Write-Error "Certificate expiration day ($expireDays) is less than 1 day.  Please provide a value greater than 1 day."
+        }
+
+        for ($createRandomCounter = 0; $createRandomCounter -lt 8; $createRandomCounter++) {
+            $randomString = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 8 |%{[char]$_})
+            $cfgFilePath = Join-Path -path $certDirPath -ChildPath "$randomString.cfg"
+            if (-Not(Test-Path -Path $cfgFilePath)) {
+                Write-Output "1" | Set-Content $cfgFilePath
+                if (Test-Path -Path $cfgFilePath) {
+                    Remove-Item $cfgFilePath
+                    break
+                }
+            } else {
+                if($createRandomCounter -gt 6) {
```

Comment:
```suggestion
                if ($createRandomCounter -gt 6) {
```

Reply:
addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014